scipy/spatial/transform/_rotation.pyx

```diff
@@ -1273,6 +1349,202 @@ cdef class Rotation:
+
+        """
+        if order in ['e', 'extrinsic']:
+            extrinsic = True
+        elif order in ['i', 'intrinsic']:
+            extrinsic = False
+        else:
+            raise ValueError("order should be 'e'/'extrinsic' for extrinsic "
+                             "sequences or 'i'/'intrinsic' for intrinsic "
+                             "sequences, got {}".format(order))
+
+        axes = np.asarray(axes)
+        if axes.ndim == 1:
+            axes = axes.reshape([1, 3])
+
+        num_axes = len(axes)
+
+        for axis in axes:
+            if len(axis) != 3:
+                raise ValueError("Axes must be vectors of length 3.")
+
+        if num_axes < 1 or num_axes > 3:
+            raise ValueError("Expected up to 3 axes, got {}".format(num_axes))
+
+        # normalize axes
+        norm = np.repeat(np.linalg.norm(axes, axis=1), 3)
+        axes = axes / norm.reshape(num_axes, 3)
+
+        if num_axes > 1 and abs(np.dot(axes[0], axes[1])) >= 1e-7:
+            raise ValueError("Consecutive axes must be orthogonal.")
+        if num_axes > 2 and abs(np.dot(axes[1], axes[2])) >= 1e-7:
+            raise ValueError("Consecutive axes must be orthogonal.")
+            
+        angles = np.asarray(angles, dtype=float)
+        if degrees:
+            angles = np.deg2rad(angles)
+
+        is_single = False
+        # Prepare angles to have shape (num_rot, num_axes)
+        if num_axes == 1:
+            if angles.ndim == 0:
+                # (1, 1)
+                angles = angles.reshape((1, 1))
+                is_single = True
+            elif angles.ndim == 1:
+                # (N, 1)
+                angles = angles[:, None]
+            elif angles.ndim == 2 and angles.shape[-1] != 1:
+                raise ValueError("Expected `angles` parameter to have shape "
+                                "(N, 1), got {}.".format(angles.shape))
+            elif angles.ndim > 2:
+                raise ValueError("Expected float, 1D array, or 2D array for "
+                                "parameter `angles` corresponding to `seq`, "
+                                "got shape {}.".format(angles.shape))
+        else:  # 2 or 3 axes
+            if angles.ndim not in [1, 2] or angles.shape[-1] != num_axes:
+                raise ValueError("Expected `angles` to be at most "
+                                "2-dimensional with width equal to number "
+                                "of axes specified, got {} for shape".format(
+                                angles.shape))
+
+            if angles.ndim == 1:
+                # (1, num_axes)
+                angles = angles[None, :]
+                is_single = True
+
+        # By now angles should have shape (num_rot, num_axes)
+        # sanity check
+        if angles.ndim != 2 or angles.shape[-1] != num_axes:
+            raise ValueError("Expected angles to have shape (num_rotations, "
+                            "num_axes), got {}.".format(angles.shape))
+
```

Comment:
Two blank lines are not recommended within a function.

Reply:
Changed